goth_tests/README.md

```diff
@@ -0,0 +1,267 @@
+# goth integration tests
+
+## Introduction
+
+This is the base directory for all of `yagna`'s integration tests which make use of [`goth`](https://github.com/golemfactory/goth) (GOlem Test Harness).
+
+`goth` is a Python package which is a backend and a runner for `yagna` integration tests. This directory is home to a Python project which defines a number of test cases and makes use of `goth` to execute them on an isolated Golem network.
+
+### Test directory structure
+Tests are organised into a directory structure based on their characteristics:
+```
+.
+├── assets                                  # Common, generated assets for tests
+│   ├── docker
+│   │   ├── yagna-goth.Dockerfile
+│   │   ├── docker-compose.yml              # Compose file for the static part of each test's network
+│   │   └── ...
+│   ├── goth-config.yml                     # Default test network configuration
+│   └── ...
+├── domain                                  # Domain-specific tests
+│   ├── exe_units
+│   ├── market
+│   ├── payments
+│   │   ├── test_zero_amount_txs.py
+│   │   └── ...
+│   └── ya-provider
+│       └── test_provider_multi_activity.py
+├── e2e                                     # End-to-end tests
+│   ├── vm
+│   │   ├── assets                          # Blender-specific assets
+│   │   │   ├── params.json
+│   │   │   ├── scene.blend
+│   │   │   └── ...
+│   │   └── test_e2e_vm.py
+│   └── wasi
+│       └── test_e2e_wasi.py
+```
+In the above structure, for each test there is a separate `test_*.py` file with a single `test_*` function defined inside.
```

Comment:
One may wonder if this is required for the test framework.

Reply:
Updated in 31cfad0